### DISCUSSION TITLE
Lambda triggered by SQS question

### DISCUSSION THREAD

**Discussion by Chandrian** [aws/aws-lambda-rust-runtime#400] [Q&A] [answered]
Hello,

I'm wondering how to manage the Records sent by the SQS service when it triggers a lambda function in Rust. In GO for example, I'm using the sqsEvent type, but could not find an equivalent for Rust. Could someone point me in the right direction? 
I tried to define my own struct with a Vec<Message>, but since Message is from a different crate, I can't make it Deserialized.
Thank you

**Comment by nmoutschen** [comment]
Hey @Chandrian !

We don't have an official crate with AWS Lambda events at the moment, but you can use the community-made `aws_lambda_events` one. It has an event type for SqsEvent: https://docs.rs/aws_lambda_events/0.5.0/aws_lambda_events/event/sqs/struct.SqsEvent.html

**Reply by Chandrian** [reply]
Thank you! Exactly what I was looking for

**Reply by seanpianka** [reply]
Are we meant to deserialize the body of the incoming HTTP request into this struct?

e.g. `let event: SqsEvent = serde_json::from_slice(event.body().to_vec().as_slice()).unwrap();`

**Reply by seanpianka** [reply]
Or, is creating a Lambda function that pulls from SQS instead of APIGW an "extension", and thus should be made with the `lambda-extension` crate (rather than `lambda-http`)?

**Reply by calavera** [reply]
@seanpianka I put an example here: https://github.com/awslabs/aws-lambda-rust-runtime/pull/560

**Reply by seanpianka** [reply]
Thank you very much!

**Comment by calavera** [comment, marked as answer]
There is an example on how to handle SQS events in the examples directory:

https://github.com/awslabs/aws-lambda-rust-runtime/blob/2461904a2821c2f69c4333cf6824344a2a2070f8/examples/basic-sqs/README.md
